dg-spark/src/main/code/org/finra/datagenerator/SparkDistributor.scala

```diff
@@ -0,0 +1,140 @@
+
+/**
+ * Take Frontiers produced by an Engine and process the Frontiers in parallel,
+ * giving the results to a DataConsumer for post-processing.
+ *
+ * Set masterURL to SparkConf using Parametrized Constructor
+ *
+ * Created by Brijesh on 6/2/2015.
+ */
+class SparkDistributor(masterURL: String) extends SearchDistributor with java.io.Serializable {
+
+  val flag: Boolean = true
+
+  val javaQueue = new util.LinkedList[util.Map[String, String]]()
+
+  //Define Spark Context and Data Consumer
+  var sparkContext: SparkContext = null
+
+  var scalaDataConsumer: ScalaDataConsumer = null
+
+  //var dataConsumer: DataConsumer = null
+  //dataConsumer = scalaDataConsumer
+
+  /**
+   * Set Data Consumer to consume output
+   *
+   * @param dataConsumer
+   * @return SearchDistributor
+   */
+  def setDataConsumer(dataConsumer: DataConsumer): SearchDistributor = {
+    //this.scalaDataConsumer = dataConsumer
+    this
+  }
+
+  /**
+   * Distribute the random number generated by Frontier from Frontier List
+   * Process data parallel using Spark - Map, Reduce method
+   *
+   * Define SparkConfiguration and SparkContext,
+   * Add dependency jar
+   *
+   * @param frontierList list of Frontiers
+   */
+  def distribute(frontierList: util.List[Frontier]): Unit = {
+
+    println("Frontier list size = " + (frontierList.size() - 1))
+
+    val conf: SparkConf = new SparkConf().setMaster(masterURL).setAppName("synchronized")
+
+    val sparkContext: SparkContext = new SparkContext(conf)
+
+    /**
+     * Add jar file to resolve dependencies
+     * NOTE: if you change something compile maven file from command line using
+     * "mvn clean package"
+     */
+    sparkContext.addJar("./dg-spark/target/dg-spark-2.2-SNAPSHOT.jar")
+
+      for (frontier <- frontierList.asScala) {
+
+        sparkContext.parallelize(1 to RandomNumberEngine.numberInEachFrontier).map {
+          i =>
+            //Generate Random Number Parallel
+            searchWorker(frontier, javaQueue, flag)
+
+            //Call Consume method for parallel processing
+            produceOutput()
+
+            0
+
+        }.reduce(_ + _)
+      }
+  }
+
+  /**
+   * While Queue is not empty
+   * Pop the elements from the Queue and give it to consume method
+   */
+  def produceOutput(): Unit = {
+
+    var lines: Int = 0
+
+    val os: OutputStream = System.out
+    val objectOS = new ObjectOutputStream(os)
+
+    while(!javaQueue.isEmpty) {
+
+      this.synchronized {
+
+        val maps = javaQueue.poll()
+
+        lines += scalaDataConsumer.consume(maps)
+
+        println()
+      }
+    }
+    println()
+  }
+
+  /**
+   * Call the searchForScenario Method which generates random number
+   *
+   * @param frontier
+   * @param javaQueue Queue
+   * @param flag boolean
+   */
+  def searchWorker(frontier: Frontier, javaQueue: util.Queue[util.Map[String, String]], flag: Boolean): Unit = {
+
+    frontier.searchForScenarios(javaQueue, null)
```

Comment:
If it's really optional, it should be defined in the method signature as flagMaybe: Option[AtomicBoolean], but if you're overriding the java method then you can't do that... I guess I don't know what the AtomicBoolean flag is for and why it's not needed here. If you can't avoid it then you can put a comment at the end to make scalastyle ignore the null:  // scalastyle:ignore